hibernate-core/src/main/java/org/hibernate/dialect/PostgreSQL81Dialect.java

```diff
@@ -171,6 +172,8 @@ public PostgreSQL81Dialect() {
 
 		registerFunction( "str", new SQLFunctionTemplate(StandardBasicTypes.STRING, "cast(?1 as varchar)") );
 
+		registerHibernateType( Types.OTHER, PostgresUUIDType.INSTANCE.getName() );
```

Comment:
Historically we have stayed away from this because lots of db datatypes get mapped to OTHER.  Is there a specific reason you changed this?

Reply:
I did it because we already register the column in the PostgreSQL Dialect:

    registerColumnType( Types.OTHER, "uuid" );

So, for consistency sake, I thought we might want to do the same for the Hibernate Type.

Reply:
This will result in [`Dialect#getTypeName(int code)`](https://github.com/hibernate/hibernate-orm/blob/master/hibernate-core/src/main/java/org/hibernate/dialect/Dialect.java#L326-L332) always returning "uuid" for all (custom) types mapped to `Types.OTHER` too right? I remember seeing some issues with this in a Union Subclass hierarchy with custom types, where the `null` values from missing properties declared in other subtypes are casted to their respective type.

Reply:
The UUID column type was mapped to Types.OTHER in Postgresql Dialect. This PR allows you to override that. Check out the associated unit tests.